store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java

```diff
@@ -102,9 +111,21 @@
  * @see Graph
  */
 public class FederatedStore extends Store {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Store.class);
+    private static final String FEDERATED_STORE_PROCESSED = "FederatedStore.processed.";
     private FederatedGraphStorage graphStorage = new FederatedGraphStorage();
     private Set<String> customPropertiesAuths;
     private Boolean isPublicAccessAllowed = Boolean.valueOf(IS_PUBLIC_ACCESS_ALLOWED_DEFAULT);
+    private static final List<Integer> ALL_IDS = new ArrayList<>();
+    private final int id;
+
+    public FederatedStore() {
+        Integer i = null;
+        while (isNull(i) || ALL_IDS.contains(i)) {
+            i = new Random().nextInt();
```

Comment:
Is it worth hoisting the Random object to stop it being created every loop, and instantiating i to a random int off the bat to prevent the need for a null check?

Reply:
99% of the time the loop will happen once.

Reply:
Is there a reason you're not assigning the value right away rather than assigning null and then checking for null?

Reply:
Because is there is collision I will need to rewrite the logic to assign the value again. duplicate line of code.

```
variable = assign value logic
check(variable)
variable = re-assign logic again // duplicate code.
```

Reply:
If you're worried about code duplication, consider using a private method. Is there a reason you're not assigning to the field directly from the get go?

Reply:
```
variable = methodAssingment()
check(variable) {
variable = methodAssingment() // still a duplicate line

methodAssingment(){ /*now there permenant  */ }
}
```

Reply:
```
variable = methodAssingment()
check(variable) {
variable = methodAssingment() // still a duplicate line

methodAssingment(){ 
    //now an extra method with a one liner sitting in the class used only once 99.99% of instantiation.
}
```